Content/Projectiles/Melee/CorruptedBattleaxeProjectile.cs

```diff
@@ -0,0 +1,169 @@
+﻿namespace PathOfTerraria.Content.Projectiles.Melee;
+
+public class CorruptedBattleaxeProjectile : ModProjectile
+{
+	private static readonly Random Random = new();
+	private bool _targetingPlayer;
+	
+	/// <summary>
+	/// The distance it will look to target the next entity when bouncing
+	/// </summary>
+	private readonly float _distance = 300f;
+
+	public override string Texture => $"{PathOfTerraria.ModName}/Assets/Items/Gear/Weapons/Battleaxe/CorruptedBattleaxe";
+        
+	public override void SetDefaults()
+	{
+		Projectile.width = 52;
+		Projectile.height = 52;
+		Projectile.friendly = true;
+		Projectile.hostile = false;
+		Projectile.penetrate = -1;
+		Projectile.tileCollide = false;
+		Projectile.ignoreWater = true;
+		Projectile.timeLeft = 300; 
+	}
+
+	public override void AI()
+	{
+		Projectile.rotation += 0.2f;
+		
+		if (_targetingPlayer)
+		{
+			Player player = FindClosestPlayer(300f);
```

Comment:
Minor inconsistency: This uses 300f, which is the only value of `_distance`, but doesn't refer to that. I'm not sure if that's intended, but it seems odd to have a variable for it and use a magic number anyway.
Same might go for the 500f when searching for nearby NPCs below, but that makes more sense.